pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -665,14 +665,9 @@ func (r *ReconcileClusterDeployment) reconcile(request reconcile.Request, cd *hi
 		return *result, err
 	}
 
-	// Sanity check the platform/cloud credentials.
-	validCreds, err := r.validatePlatformCreds(cd, cdLog)
-	if err != nil {
-		cdLog.WithError(err).Error("unable to validate platform credentials")
-		return reconcile.Result{}, err
-	}
-	// Make sure the condition is set properly.
-	_, err = r.setAuthenticationFailure(cd, validCreds, cdLog)
+	// Sanity check the platform/cloud credentials and set hivev1.AuthenticationFailureClusterDeploymentCondition
+	validCreds, authError := r.validatePlatformCreds(cd, cdLog)
+	_, err := r.setAuthenticationFailure(cd, validCreds, authError, cdLog)
 	if err != nil {
```

Comment:
If the status update succeeds, but `authError` is not `nil`, I think you want to `return reconcile.Result{}, authError`.

Reply:
That happens right below this block of code. We were always skipping it before.

Reply:
It is based on the condition having been set but could check `authError` instead.

Reply:
Ah, I see it now. Yeah, that reads kind of weird, but :man_shrugging: